public/global.js

```diff
@@ -51,7 +58,19 @@ const onFillVarsChange = (editor) => {
 };
 
 const onFillVarsSave = () => {
-  previousFillVars = window.fillVarsCodeMirrorEditor.getValue();
+  previousFillVarsText = window.fillVarsCodeMirrorEditor.getValue()
+  
+  const lcFillVars = parseJSONText(localStorage.getItem("fillVars"));
+
+  const newLcFillVars = {
+    ...lcFillVars,
+    [templateName]: {
+      ...lcFillVars[templateName],
+      ...parseJSONText(previousFillVarsText)
+    }
+  };
+
+  localStorage.setItem("fillVars", JSON.stringify(newLcFillVars));
```

Comment:
Wouldn't it be better to save this variable separately per template? If a user is switching between 2 templates, I think it is better to save 2 different variable sets, one for each template. One way is to prefix the local storage variable name with the template name, or have one variable but maintain a tree of variables where the root elements are the template names.

Reply:
After some thoughts, I think this is not necessary. Even with different templates, usually merge variables would be very similar and it is even better to save one set of variables.

Reply:
Ah, I just tried it and realized you already save the template name within the local storage variable, that's ok, we will leave it here.